airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/write/WriteOperation.kt

```diff
@@ -74,9 +77,21 @@ class WriteOperation(
 @Factory
 class InputStreamProvider {
     @Singleton
-    @Secondary
-    @Requires(property = Operation.PROPERTY, value = "write")
-    fun make(): InputStream {
-        return System.`in`
+    @Named("inputStream")
+    fun makeWrite(
+        @Value("\${${Operation.PROPERTY}}") operation: String,
+        @Named("checkInputStreamPipe") pipe: PipedOutputStream?,
+    ): InputStream =
+        if (operation == "check") {
+            PipedInputStream(pipe!!)
+        } else {
+            System.`in`
+        }
+
+    @Singleton
+    @Named("checkInputStreamPipe")
+    @Requires(property = Operation.PROPERTY, value = "check")
+    fun makeCheckPipe(): PipedOutputStream {
+        return PipedOutputStream()
```

Comment:
I could be convinced that it's better to just have the DestinationChecker do `val pipe = PipedOutputStream(inputStream as PipedInputStream)`, and then we can kill this bean?

Reply:
Unless this is doing something for testability or what have you, axing it seems good.

Reply:
done in https://github.com/airbytehq/airbyte/pull/59691/commits/424ef26a4a5ef4ee802f38d23f68fb6eb4df7bfd